dll/win32/browseui/shellbrowser.cpp

```diff
@@ -2257,12 +2311,35 @@ HRESULT STDMETHODCALLTYPE CShellBrowser::TranslateAcceleratorSB(MSG *pmsg, WORD
 
 HRESULT STDMETHODCALLTYPE CShellBrowser::BrowseObject(LPCITEMIDLIST pidl, UINT wFlags)
 {
-    if ((wFlags & SBSP_EXPLOREMODE) != NULL)
+    UINT newbrowser = GetNewBrowserFlag(wFlags);
```

Comment:
Why using a separate variable (newbrowser) while it _could_ be possible to reuse wFlags there?
Your `GetNexBrowserFlags()` appears, in a sense, to sanitize the `wFlags` and return it supplemented with extra flags. Only in the case where both `SBSP_NAVIGATEBACK` and `SBSP_NAVIGATEFORWARD` are set together, a "hardcoded" value is returned. But in that latter case, I could ask, why not instead removing these flags from `wFlags` and set the `SBSP_SAMEBROWSER` in it instead?

Reply:
It is only supposed to tell you about SBSP_SAMEBROWSER vs SBSP_NEWBROWSER.

 > Only in the case where both SBSP_NAVIGATEBACK and SBSP_NAVIGATEFORWARD are set together

That is incorrect, it happens if one of them (or both) is set.

Reply:
> It is only supposed to tell you about SBSP_SAMEBROWSER vs SBSP_NEWBROWSER.

```c
UINT CShellBrowser::GetNewBrowserFlag(UINT Flags)
{
    if ((Flags & (SBSP_SAMEBROWSER | SBSP_NEWBROWSER)) == SBSP_DEFBROWSER)
    {
        if (!fCurrentDirectoryPIDL || IsControlWindowShown(FCW_TREE, NULL) == S_OK)
            Flags |= SBSP_SAMEBROWSER; // Force if this is the first navigation or the folder tree is present
        else
            Flags |= (!!gCabinetState.fNewWindowMode) ^ (GetAsyncKeyState(VK_CONTROL) < 0) ? SBSP_NEWBROWSER : SBSP_SAMEBROWSER;
    }
    if (Flags & (SBSP_NAVIGATEBACK | SBSP_NAVIGATEFORWARD))
        return SBSP_SAMEBROWSER; // Force same browser for now
    return Flags;    ///////// <---------------- See comment/question below
}
```
(Comment/question): When this `return Flags` is taken, and the `if ((Flags & (SBSP_SAMEBROWSER | SBSP_NEWBROWSER)) == SBSP_DEFBROWSER)` branch has been taken, it seems that the returned value can contain _much more_ than the samebrowser/newbrowser flags: in particular the value will contain also all the other SBSP flags that may have been set by the caller into this flags variable.

Obviously, the `Flags |= ...` have also enforced setting the samebrowser or newbrowser flags in there.

This is in _**that case**_ that I was saying that, in a sense, the function was sanitizing the flags and returned them.
The only case where it doesn't return any of the possible other original flags the caller may have set, is the `        return SBSP_SAMEBROWSER; // Force same browser for now` case.
Because of this, I was wondering whether that last case could be adapted, in order to do something in `Flags`, instead (e.g. setting `SBSP_SAMEBROWSER` and removing `SBSP_NEWBROWSER` to be sure) then returning the modified `Flags` variable (that could include other caller-set flags besides those modified in that function).